src/libraries/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -129,15 +129,6 @@ public static ConsoleKeyInfo ReadKey(bool intercept)
             bool previouslyProcessed;
             ConsoleKeyInfo keyInfo = StdInReader.ReadKey(out previouslyProcessed);
 
-            // Replace the '\n' char for Enter by '\r' to match Windows behavior.
-            if (keyInfo.Key == ConsoleKey.Enter && keyInfo.KeyChar == '\n')
```

Comment:
I don't think so. Fundamentally the problem lies in the native layer which translates CR characters to LF. This might be desirable in `Read()`/`ReadLine()` scenaria, however it clearly shouldn't be in the case of `ReadKey()`. Once the issue in the native layer has been addressed, this check is no longer required.